src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,18 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  const foundKey = Object.keys(config.carry.sizes).reverse()
+    .find(key => (key <= Game.rooms[this.memory.base].energyCapacityAvailable));
+  let carryCapacity = config.carry.sizes[foundKey][1] * CARRY_CAPACITY;
+
+  const workParts = this.body.filter(part => part.type === WORK).length;
```

Comment:
we can ally looping until we encounter a different part type as they are sorted https://github.com/TooAngel/screeps/blob/carrysourcer/src/prototype_room_creepbuilder.js#L307
depending if we want to definitly sort them or still allow to unsort. but we still can place them at start or end.